app/models/miq_cockpit_ws_worker/authenticator.rb

```diff
@@ -0,0 +1,67 @@
+class MiqCockpitWsWorker::Authenticator
+  def authenticate_for_host(token, host)
+    user_obj = user_for_token(token)
+    return {} unless user_obj
+
+    # is it a container deployment
+    creds = find_container_node_creds(user_obj, host)
+
+    # Is it a VM
+    creds = find_vm_creds(user_obj, host) if creds.nil?
+
+    {
+      :valid  => true,
+      :known  => !creds.nil?,
+      :key    => creds.nil? ? nil : creds.auth_key,
+      :userid => creds.nil? ? nil : creds.userid,
```

Comment:
this is the same as `creds.try(:userid)` and same for `auth_key`.  If It's nil, try, returns nil without raising an exception.